src/evm/protocol/uniswap_v4/hooks/hook_handler_creator.rs

```diff
@@ -44,8 +44,45 @@ pub trait HookHandlerCreator: Send + Sync {
     ) -> Result<Box<dyn HookHandler>, InvalidSnapshotError>;
 }
 
+pub struct GenericVMHookHandlerCreator;
+
+impl HookHandlerCreator for GenericVMHookHandlerCreator {
+    fn instantiate_hook_handler(
+        &self,
+        _params: HookCreationParams,
+    ) -> Result<Box<dyn HookHandler>, InvalidSnapshotError> {
+        todo!()
+    }
+}
+
 // Workaround for stateless decoder trait.
+// Mapping from hook address to the handler creator.
 lazy_static! {
     static ref HANDLER_FACTORY: RwLock<HashMap<Address, Box<dyn HookHandlerCreator>>> =
         RwLock::new(HashMap::new());
 }
+
+lazy_static! {
+    static ref DEFAULT_HANDLER: RwLock<Box<dyn HookHandlerCreator>> =
+        RwLock::new(Box::new(GenericVMHookHandlerCreator {}));
+}
+
+pub fn register_hook_handler(
+    hook: Address,
+    handler: Box<dyn HookHandlerCreator>,
+) -> Result<(), SimulationError> {
+    HANDLER_FACTORY
+        .write()
+        .map_err(|e| SimulationError::FatalError(e.to_string()))?
+        .insert(hook, handler);
+    Ok(())
+}
+
+pub fn set_default_hook_handler(
```

Comment:
I am having trouble understanding when we would ever need to set this. I am assuming we don't, since it's already set to `GenericVMHookHandlerCreator` at start time?

Reply:
ah if you are doing the `DEFAULT_HANDLER`, then yes we don't need this!